admin.py

```diff
@@ -25,3 +37,55 @@ def decorate(self, *args, **kwargs):
 
   return decorate
 
+
+def RequireAppAndDomainAdmin(oauth_decorator):
+  """Decorator to require the user to be an admin."""
+  def Wrap(func):
+    """Decorator to require the user to be an admin."""
+    def decorate(self, *args, **kwargs):
+      """Actual decorate function that requires admin.
+
+      Args:
+        args: Parameters passed on to the specified function if successful.
+        kwargs: Parameters passed on to the specified function if successful.
+      """
+      user = users.get_current_user()
+      if not user or user is None:
+        logging.error('User is not logged in.')
+        self.abort(403)
+
+      identifier = user.email()
+      if identifier is None or identifier is '':
+        logging.error('No identifier found for the user.')
+        self.abort(403)
+
+      directory_user = None
+      try:
+        directory_service = GoogleDirectoryService(oauth_decorator)
+        directory_user = directory_service.GetUser(identifier)
+      except errors.HttpError:
+        logging.error('Exception when asking dasher for this user.')
+        self.abort(403)
+
+      if not directory_user or directory_user is None:
+        logging.error('User was not found in dasher.')
+        self.abort(403)
+      if not directory_user['isAdmin']:
+        logging.error('User is not a dasher admin.')
+        self.abort(403)
+      return func(self, *args, **kwargs)
+
+    return decorate
+  return Wrap
+
+
+OAUTH_ALL_SCOPES_DECORATOR = OAuth2Decorator(
+    client_id=OAuth.GetOrInsertDefault().client_id,
+    client_secret=OAuth.GetOrInsertDefault().client_secret,
+    scope=SCOPES)
+
+OAUTH_USER_SCOPE_DECORATOR = OAuth2Decorator(
```

Comment:
Is there a reason to break out the user scope into a separate decorator?  I can see this can help to tighten access.  But this would prompt the user a couple of times to approve the oauth access?  I read somewhere (I can't find it now) that the better practice is for the app to prompt the user once to authorize all the scopes that the app needs.  And maintaining separate decorators seem to be making the code more complicated.  Would it be possible to just maintain one oauth decorator?